pkg/scheduler/flavorassigner/tas_flavorassigner.go

```diff
@@ -152,26 +158,24 @@ func checkPodSetAndFlavorMatchForTAS(cq *cache.ClusterQueueSnapshot, ps *kueue.P
 			// Skip flavors which don't have the requested level
 			return ptr.To(fmt.Sprintf("Flavor %q does not contain the requested level", flavor.Name))
 		}
-	}
-	// If this is a TAS-only CQ, then no TopologyRequest is ok
-	if isTASImplied(ps, cq) {
+		// PodSet requires TAS and the flavor supports it, so it's a match.
 		return nil
 	}
-	// For PodSets which don't use TAS skip resource flavors which are only for TAS
-	if ps.TopologyRequest == nil && flavor.Spec.TopologyName != nil {
+	// PodSet doesn't require TAS, but the flavor supports it.
+	if flavor.Spec.TopologyName != nil {
 		return ptr.To(fmt.Sprintf("Flavor %q supports only TopologyAwareScheduling", flavor.Name))
 	}
+	// PodSet doesn't require TAS and the flavor doesn't support it, so it's a match.
 	return nil
 }
 
-// isTASImplied returns true if TAS is requested implicitly - there is no
-// explicit
+// isTASImplied returns true if TAS is requested implicitly
 func isTASImplied(ps *kueue.PodSet, cq *cache.ClusterQueueSnapshot) bool {
-	return ps.TopologyRequest == nil && cq.IsTASOnly()
+	return !workload.IsTASExplicitlyRequested(ps) && cq.IsTASOnly()
 }
 
 // isTASRequested checks if TAS is requested for the input PodSet, either
 // explicitly or implicitly.
 func isTASRequested(ps *kueue.PodSet, cq *cache.ClusterQueueSnapshot) bool {
-	return ps.TopologyRequest != nil || cq.IsTASOnly()
+	return workload.IsTASExplicitlyRequested(ps) || isTASImplied(ps, cq)
```

Comment:
Uhm, it seems that we have some way to check if cq has TAS only flavors. (`cq.IsTASOnly()` vs `isTASImplied()`). I guess that we can remove the `isTASImplied` function.
We should refactor that in a follow-up.

Reply:
IIUC, from now, there is no difference between explicitly unconstrained and implied TAS, right?
In that case, could we build topology request for unconstrained when all cq flavors have topology in https://github.com/kubernetes-sigs/kueue/blob/1b12701b63a3b32047a4ffd10b4b28d77b282f11/pkg/controller/jobframework/tas.go#L69-L74